orangecontrib/spectroscopy/widgets/gui.py

```diff
@@ -474,3 +477,152 @@ def set_default(self, v):
     def focusInEvent(self, *e):
         self.focusIn.emit()
         return QWidget.focusInEvent(self, *e)
+
+
+class VerticalPeakLine(pg.InfiniteLine):
+
+    """pyqtgraph.InfiniteLine with adjustments for spectra analysis
+    """
+
+    def __init__(self, pos=None, angle=90, pen=None, movable=True,
+                 bounds=None, label=None, span=None):
+        super().__init__(pos, angle, pen, movable, bounds)
+        self.moving = False
+        self.mouseHovering = False
+        self.span = span
+        self._boundingRect = None
+        self._endPoints = [0, 1]
+        self._bounds = None
+        self._lastViewSize = None
+
+        if label is None:
+            self.label = VerticalPeakLineLabel(self,
+                                               text=str(self.getXPos()), position=(1))
+
+    def setSpan(self, mn, mx):
+        if self.span != (mn, mx):
+            self.span = (mn, mx)
+            self.update()
+
+    def _computeBoundingRect(self):
+        vr = self.viewRect()
+        if vr is None:
+            return QtCore.QRectF()
+
+        px = self.pixelLength(direction=Point(1, 0), ortho=True)
+        if px is None:
+            px = 0
+        pw = max(self.pen.width() / 2, self.hoverPen.width() / 2)
+        w = max(4, 0 + pw) + 1
+        w = w * px
+        br = QtCore.QRectF(vr)
+        br.setBottom(-w)
+        br.setTop(w)
+        left = self.span[0]
+        right = self.span[1]
+        #sets our line span to go from our minimum value to maximum
+        br.setLeft(left)
+        br.setRight(right)
+        br = br.normalized()
+
+        vs = self.getViewBox().size()
+        if self.mouseDragEvent:
+            left = self.span[0] - self.getYPos()
+
+        if self._bounds != br or self._lastViewSize != vs:
+            self._bounds = br
+            self._lastViewSize = vs
+            self.prepareGeometryChange()
+
+        self._endPoints = (left, right)
+        self._lastViewRect = vr
+
+        return self._bounds
+
+    def boundingRect(self):
+        if self._boundingRect is None:
+            self._boundingRect = self._computeBoundingRect()
+        return self._boundingRect
+
+    def paint(self, p, *args):
+        p.setRenderHint(p.Antialiasing)
+        left, right = self._endPoints
+        pen = self.currentPen
+        pen.setJoinStyle(QtCore.Qt.MiterJoin)
+        p.setPen(pen)
+        p.drawLine(Point(left, 0), Point(right, 0))
+
+
+    def mouseDragEvent(self, ev):
```

Comment:
Should not be modified. The only added thing is `updateLabel` could be connected to the sigDragged signal in `__init__`.